app/scripts/controllers/registration.js

```diff
@@ -37,26 +37,33 @@ angular
     $scope.savingAnswers = false;
 
     $scope.activePageId = pageId || '';
-    $scope.page = _.find(conference.registrationPages, { id: pageId });
-    $scope.activePageIndex = _.findIndex($scope.conference.registrationPages, {
-      id: pageId,
-    });
     // Filter through all pages and remove any empty pages
     // Check for registrant needed since on the welcome page
     // user could possibly not have started registration yet
     $scope.validPages =
       $scope.currentRegistration.registrants.length >= 1
         ? $scope.conference.registrationPages.filter(page => {
+            const registrantId = $scope.currentRegistrant
+              ? $scope.currentRegistrant
+              : $scope.currentRegistration.registrants[0].id;
+
             return (
-              page.blocks.filter(
-                block =>
-                  !block.registrantTypes.includes(
-                    $scope.currentRegistration.registrants[0].registrantTypeId,
+              page.blocks.filter(block =>
+                validateRegistrant.blockVisible(
```

Comment:
I should have realized there was a function that existed that was doing what I was trying to do. So I went ahead and utilized that instead.